gplugins/lumerical/tests/test_lumerical_utils.py

```diff
@@ -0,0 +1,267 @@
+                info={},
+                background_doping_concentration=100000000000000.0,
+                background_doping_ion="p",
+                orientation="100",
+            ),
+            "n++": LayerLevel(
+                name="n++",
+                layer=(5, 0),
+                thickness=0.22,
+                zmin=0,
+                layer_type="doping",
+                background_doping_concentration=1e17,
+                background_doping_ion="n",
+            ),
+        }
+    )
+
+    # Check process file in Lumerical MODE, FDTD, and CHARGE
+    try:
+        import lumapi
+    except Exception as err:
+        raise AssertionError(
+            f"{err}\nUnable to import lumapi. Check sys.path for location to lumapi.py."
+        ) from err
+
+    message = ""
+    sessions = [
+        lumapi.MODE(hide=not DEBUG_LUMERICAL),
+        lumapi.FDTD(hide=not DEBUG_LUMERICAL),
+        lumapi.DEVICE(hide=not DEBUG_LUMERICAL),
+    ]
+    for s in sessions:
+        success = False
+
+        # Create passive LBR process file
+        process_file_lumerical2021 = layerstack_to_lbr(
+            layer_map, layerstack=layerstack_lumerical2021
+        )
+        try:
+            s.addlayerbuilder()
+            s.loadprocessfile(str(process_file_lumerical2021))
+            success = success or True
+            message += f"\nSUCCESS ({type(s)}): Passives only process file successfully loaded."
+        except Exception as err:
+            success = success or False
+            message += f"\n{err}\nWARNING ({type(s)}): Passives only process file failed to load."
+
+        if ENABLE_DOPING:
+            # Create passive and dopant LBR process file
+            process_file_lumerical2023 = layerstack_to_lbr(
+                layer_map, layerstack=layerstack_lumerical2023
+            )
+
+            try:
+                s.addlayerbuilder()
+                s.loadprocessfile(str(process_file_lumerical2023))
+                success = success or True
+                message += f"\nSUCCESS ({type(s)}): Passives and dopants process file successfully loaded."
+            except Exception as err:
+                success = success or False
+                message += f"\nWARNING ({type(s)}): {err}\nPassives and dopants process file failed to load. Lumerical 2021 version does not support dopants."
+        s.close()
+
+        # If process file cannot be imported into particular Lumerical simulator, raise error
+        if not success:
+            raise AssertionError(f"Process file cannot be imported into {type(s)}")
+
+    if success:
+        message += (
+            "\nTest passes if one of passives only or passives + dopants process files work."
+            + "\nThis is used to capture Lumerical versions 2021 that only work with passive only process files "
+            + "and subsequent versions of Lumerical that support both passives and dopants."
+        )
+    logger.info(message)
+    assert success, message
```

Comment:
ideally we have some functional verification (import component + layerstack, run sim and make sure it converges or something) might be able to have some reuse with the corresponding designrecipe tests

Reply:
Yes, this is done via test_eme.py and test_fdtd.py. Here, the purpose is a unit test for importing process file since the process file creation is reversed engineered from the output of Lumerical's software. If the process file format changes in the future, we can catch it here.